src/dafny/Relations.md

```diff
@@ -0,0 +1,36 @@
+
+## The `Relations` module {#sec-relations}
+
+The `Relations` module defines a number of properties of functions.
+
+Properties of binary predicates:
+- Reflexive: `R(x,x)` is true
+- Irreflexive: `R(x,x)` is false
+- AntiSymmetric: `R(x,y) && R(y,x) ==> x==y`
+- Transitive : `R(x,y) && R(y,z) ==> R(x,z)`
+- Connected : `x != y ==> R(x,y) || R(y,x)`
+- StronglyConnected : `R(x,y) || R(y,x)`
+- TotalOrdering : Reflexive, AntiSymmetric, Transitive, StronglyConnected (e.g., `<=` on integers)
+- StrictTotalOrdering : Irreflexive, AntiSymmetric, Transitive, Connected (e.g., `<` on integers)
+
+A property of unary functions:
+- Injective : `f(x) == f(y) ==> x == y`
+
+These properties are sometimes required for other functions. For example, 
+if one wants to sort a sequence by some relation `R`, one must establish that `R` is a _Total Ordering_. 
+In fact, that is part of the precondition of the sorting function.
```

Comment:
This is slightly tricky since the one sorting function we have (Arrays.BinarySearch) requires a strict total ordering, but #57 will require a total ordering. Ideally we'd clean that up without breaking people, but for now the line above should probably say "Strict Total Ordering" instead.

Reply:
Revised